### PR TITLE
caliper: Add v2.12.1

### PR DESCRIPTION
**Pull request by daboehme**

### CODE REVIEW

**Comment by tldahlgren** [comment]
Problems with one stack's generation.

@spackbot run pipeline

**Review by tldahlgren — DISMISSED** [review]
Confirmed the sha256.

**Comment by zackgalbreath** [comment]
I'm looking at the GitLab side to get this pipeline (hopefully) passing.

**Comment by zackgalbreath** [comment]
Your GitLab pipeline issues should be fixed now, but it looks like they did detect a legitimate failure to apply a patch file.

https://gitlab.spack.io/spack/spack/-/jobs/14171349#L461

**Comment by daboehme** [comment]
> Your GitLab pipeline issues should be fixed now, but it looks like they did detect a legitimate failure to apply a patch file.
> 
> https://gitlab.spack.io/spack/spack/-/jobs/14171349#L461

Thanks! The latest version actually fixes this issue so we don't need the patch anymore, but I missed this in the spack recipe. Should be fixed now.

File: var/spack/repos/builtin/packages/caliper/package.py [hunk 1]
```diff
@@ -104,7 +109,7 @@ class Caliper(CachedCMakePackage, CudaPackage, ROCmPackage):
     variant("tools", default=True, description="Enable tools")
 
     # TODO change the 'when' argument for the next release of Caliper
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/caliper/package.py]: Nit: Do you still need this comment?

**Review by tldahlgren — DISMISSED** [review]
LGTM. Do you want to remove the unnecessary comment before this is merged?

File: var/spack/repos/builtin/packages/caliper/package.py (hunk 1 above)
Comment (daboehme) [reply var/spack/repos/builtin/packages/caliper/package.py]: Good point. I've removed the comment.

**Review by tldahlgren — APPROVED** [review]
Thanks.

**Pull request merged**